src/include/btmem.h

```diff
@@ -758,6 +773,23 @@ struct __wt_page {
 
     uint64_t cache_create_gen; /* Page create timestamp */
     uint64_t evict_pass_gen;   /* Eviction pass generation */
+
+#ifdef HAVE_DIAGNOSTIC
```

Comment:
Did you explore having two different macros rather than wrapping uses of the macro in `ifdef HAVE_DIAGNOSTIC`?

i.e:
```
#ifdef HAVE_DIAGNOSTIC
#define WT_SPLIT_PAGE_SAVE_STATE <code as it is in PR>
#else
#define WT_SPLIT_PAGE_SAVE_STATE
#endif
```

Reply:
That's a good point, we can definitely do that. @etienneptl If we do this should we only enable it for the `recovery stress test` evergreen task?

Reply:
I don't think we should only enable the code for a specific test. Other tests may trigger split issues and this information would be valuable.

Reply:
@agorrod Is there anything worth adding in non-diagnostic mode? This is only to gather more information about splits when we encounter a split generation bug in our testing.

Reply:
Nevermind! I realise this is how we `#define` other macros to avoid repeatedly calling `#ifdef HAVE_DIAGNOSTIC`. I'll make that change.